src/validation.cairo

```diff
@@ -56,8 +58,30 @@ fn next_prev_timestamps(self: @ChainState, block: @Block) -> Span<u32> {
 }
 
 fn compute_total_work(self: @ChainState, block: @Block) -> u256 {
-    // TODO: implement
-    *self.total_work
+    let block_target: u256 = convert_u32_to_u256(*block.header.bits);
+    let work_in_block: u256 = compute_work_from_target(block_target);
+    let new_total_work: u256 = *self.total_work + work_in_block;
+    new_total_work
+}
+
+fn convert_u32_to_u256(val: u32) -> u256 {
+    let felt_val: felt252 = val.into();
+    let mut val_u256: u256 = 1;
+    let low: u128 = felt_val.try_into().unwrap();
+    let high: u128 = 0;
+    val_u256.low = low;
+    val_u256.high = high;
+    val_u256
+}
+
+// Need to compute 2**256 / (target+1), but we can't represent 2**256
+// as it's too large for an u256. However, as 2**256 is at least as large
+// as target+1, it is equal to ((2**256 - target - 1) / (target+1)) + 1,
+// or ~target / (target+1) + 1.
+fn compute_work_from_target(target: u256) -> u256 {
+    let div: u256 = ~target / (target + ONE_256);
```

Comment:
why not just: `~target / (target + 1_u256) + 1_u256`. I mean just this should be the function body.